test/uk/gov/hmrc/tai/viewModels/TaxCodeDescriptorSpec.scala

```diff
@@ -16,33 +16,25 @@
 
 package uk.gov.hmrc.tai.viewModels
 
-import controllers.actions.FakeValidatePerson
 import controllers.i18n.TaiLanguageController
-import controllers.{FakeAuthAction, FakeTaiPlayApplication}
-import mocks.{MockPartialRetriever, MockTemplateRenderer}
-import org.scalatest.mockito.MockitoSugar._
-import org.scalatestplus.play.PlaySpec
-import play.api.i18n.{I18nSupport, Lang, Messages, MessagesApi}
-import uk.gov.hmrc.domain.Generator
-import uk.gov.hmrc.play.partials.FormPartialRetriever
-import uk.gov.hmrc.tai.config.ApplicationConfig
+import play.api.i18n.{Lang, Messages}
+import play.api.test.Helpers.stubControllerComponents
 import uk.gov.hmrc.tai.model.domain._
 import uk.gov.hmrc.tai.model.domain.income.{Live, OtherBasisOfOperation, TaxCodeIncome, Week1Month1BasisOfOperation}
 import uk.gov.hmrc.tai.service.PersonService
 import uk.gov.hmrc.urls.Link
+import utils.BaseSpec
 
 import scala.collection.immutable.ListMap
 
-class TaxCodeDescriptorSpec extends PlaySpec with FakeTaiPlayApplication with I18nSupport {
-
-  implicit val messagesApi: MessagesApi = app.injector.instanceOf[MessagesApi]
+class TaxCodeDescriptorSpec extends BaseSpec {
 
   "standAloneTaxCodeExplanation" must {
     "return the correct explanation" when {
       Seq("0T", "BR", "NT").foreach(taxCode => {
         s"tax code is $taxCode" in {
           TaxCodeDescriptorConcrete
-            .describeTaxCode(taxCode, OtherBasisOfOperation, Map.empty[String, BigDecimal]) mustBe
+            .describeTaxCode(taxCode, OtherBasisOfOperation, Map.empty[String, BigDecimal], true, appConfig) mustBe
```

Comment:
@anand462 @vimalsuresh need to check over the tax code descriptions are correct for welsh, scottish and rUK users

Reply:
Tested and validated.